src/data_tools/src/benchmark.cpp

```diff
@@ -321,20 +444,63 @@ void track_error_benchmark::add_ground_truth(mbes_ping::PingsT& pings)
     track_img_path = dataset_name + "_benchmark_track_img.png";
 }
 
+void track_error_benchmark::add_ground_truth(PointsT& map_points, PointsT& track_points){
+
+    int rows = 500;
+    int cols = 500;
+
+    for(const Eigen::MatrixXd& track_i: track_points){
+        for(unsigned int i=0; i<track_i.rows(); i++){
+            gt_track.push_back(track_i.row(i));
+        }
+    }
+    track_img_params(map_points, rows, cols);
+    add_benchmark(map_points, track_points, "ground_truth");
+    track_img_path = dataset_name + "_benchmark_track_img.png";
+}
+
 void track_error_benchmark::add_initial(mbes_ping::PingsT& pings)
 {
     input_pings = pings;
 }
 
+void track_error_benchmark::add_benchmark(PointsT& maps_points, PointsT& tracks_points,
+                                          const std::string& name){
+    cv::Mat error_img;
+    Eigen::MatrixXd error_vals;
+    double consistency_rms_error;
+    std::vector<std::vector<std::vector<Eigen::MatrixXd>>> grid_maps = create_grids_from_matrices(maps_points);
```

Comment:
I like how you did this to enable using the same consistency thing for both. Again, ">>>".